### DISCUSSION TITLE
minimal depth?

### DISCUSSION THREAD

**Discussion by Wonmo** [sebp/scikit-survival#162] [Q&A]
First of all, I appreciate all of your great works on scikit-survival, Dr. Pölsterl.

Is there any option to calculate minimal depth for each variable?

I've read your suggestion to [calculate feature importance ](https://scikit-survival.readthedocs.io/en/stable/user_guide/random-survival-forest.html)using eli5. However, to measure feature importance, many research articles calculate the minimal-depth, which is the shortest distance from the tree trunk to the branch level of the first split of the variable. For example, [this paper](https://jnm.snmjournals.org/content/59/5/769.short). Do you have any plan to implement it? or any hints using API on scikit-survival?

Please feel free to share your opinion.

**Comment by sebp** [comment]
There's currently no function that directly gives you the average depth for each feature, but the raw information is available. RandomSurvivalForest has an [estimators_](https://scikit-survival.readthedocs.io/en/latest/api/generated/sksurv.ensemble.RandomSurvivalForest.html#sksurv.ensemble.RandomSurvivalForest.estimators_) attribute, which is a list of SurvivalTree instances. Each tree instance in turn has a [tree_]( https://scikit-survival.readthedocs.io/en/latest/api/generated/sksurv.tree.SurvivalTree.html#sksurv.tree.SurvivalTree.tree_) attribute that describes the tree and can be used to compute average depth. There's a [sklearn guide](https://scikit-learn.org/stable/auto_examples/tree/plot_unveil_tree_structure.html#sphx-glr-auto-examples-tree-plot-unveil-tree-structure-py) that describes the tree structure and how to traverse it.

**Reply by Wonmo** [reply]
Thank you so much, @sebp!
